contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,248 @@
+
+1 - A node local volume attached to the container
+
+2 - A persistent volume claim from the cloud provider.
+
+Either of the 2 options above can be used/specified by the user to attach a persistent store to the pods.
+
+It is known, that as of now, the same persistent volume, even if can be used in a different cluster, k8s directly does not provide an API, yet, which can aid the same. In the absence of no direct way of quick migration of the storage data from one zone to another, or from one cloud provider to another if the case be, the proposal is to disallow migration of pods across clusters.
+
+## Scale up/Scale down
+
+When a federated statefulset is scaled up or down, the distribution of replicas will be recalculated and the replica numbers in each cluster will be updated independently.
+
+The behaviour of the scaling on being targeted using a **federated hpa** will be discussed in a separate design.
+
+## Handling additional LB services
+
+As listed in the approaches, the current option of exposing the statefulset pods across clusters is to assign a LB service to each replica pod.
+If the federated service controller is used (by creating the service in federation) to create these services, then as per the current behaviour the service will be created in each cluster. 
+This leads to exactly one unused service per statefulset per cluster.
+There are two alternatives to circumvent this problem:
+
+1 - Cluster affinity is introduced at least for services such that the controller creates the service only in needed cluster. (The need of cluster affinity and anti-affinity in general is discussed [elsewhere](https://github.com/kubernetes/kubernetes/issues/41442) also)
+
+2 - Federated statefulset controller handles the creation of cluster local services, rather than the federated service controller. It also will need to handle the dns records for each LB service into the federation (or cloud provider specific) public dns server.
+
+We propose using alternative 1 listed here, as it fits broader scheme of things and is more consistent with the user expectation of being able to query all needed resources from the federation control plane and is less confusing to use at the same time.
+
+# Conclusion
+
+Alternative 1 can solve only 1 of the listed use cases (scenario 1), and does not remain consistent with the existing federated controllers.
+
+Alternative 2 can solve both the use cases, is consistent with the existing federated controllers and makes more sense in the scheme of things.
+
+# Limitations (to be discussed)
+
+_1 – In in case of a particular replica or the cluster containing particular replicas die, what should be the behavior._
+
+It can be resurrected with the same identity in the original cluster only
+
+_2 – Is there a migration scenario? If there is, in case a replica needs migration, how do we handle this?_
+
+elaborated by @quinton-hoole
+
+Strictly speaking, migration between clusters in the same zone is quite feasible.
```

Comment:
More detail is still required here.  If storage can be replicated between clusters (at least those hosted by the same infrastructure e.g. gce), then including the cluster name in the identity doesn't seem like such a good idea.